rusoto/core/src/signature.rs

```diff
@@ -605,4 +607,12 @@ mod tests {
         assert_eq!("/path%20with%20spaces%3A%20the%20sequel%2B%2B",
                    canonical_uri_string);
     }
+
+    #[test]
+    fn signature_generation() {
```

Comment:
I wrote these tests against the old `ring`-based implementation, and then ported them, to ensure that I don't break any of the subtle details of the signing logic.